internal/config/config.go

```diff
@@ -170,44 +170,68 @@ func For(resources ClusterResources, validate Validate) (*Config, error) {
 		return nil, err
 	}
 
-	cfg := &Config{
-		Pools:       map[string]*Pool{},
-		BFDProfiles: map[string]*BFDProfile{},
+	cfg := &Config{}
+
+	cfg.BFDProfiles, err = parseBFDProfiles(resources)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Peers, err = parsePeers(resources, cfg.BFDProfiles)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Pools, err = parsePools(resources)
+	if err != nil {
+		return nil, err
 	}
 
+	return cfg, nil
+}
+
+func parseBFDProfiles(resources ClusterResources) (map[string]*BFDProfile, error) {
```

Comment:
I'd reuse the same "for*" pattern in naming

Reply:
done